Building out carousel component

jondionowens opened the pull request:
First basic implementation of the carousel component that gets activated when you click on one of the images in the grid view.

**Next steps:**
1) Make the carousel auto select the image that was clicked in the grid
2) Build out thumbnail selector like the one on airbnb.com
3) Address the jankiness of the "forceStateUpdate" function

Commit:
Hard coded service location into API calls, so things work on proxy

Commit:
Updated conditional to make requests without id default to id 1

Commit:
Started general frame work for carousel components

Commit:
Deleted 'example' directory

Commit:
Pulling in a test photo to start building primary photo component

Commit:
[WIP] Working through carousel component

Commit:
Merge branch 'carousel' of https://github.com/rpt09-mulder/gallery into carousel

Commit:
Converted photo object in carousel to an array using lodash.
- [WIP] working on getting arrow sliding working

Commit:
Set up beginnings of click handler for slider

Commit:
Arrow sliders now working

Commit:
Implemented exit button on carousel
- Also turn the carousel toggle on

Commit:
Slight modifications to the carousel structure

Commit:
Implemented background grow animation on photo cells
- Various other css optimizations

Commit:
Fixed hiding bug and removed unused id's from photos

insivika commented on client/components/App.css:
```diff
@@ -27,17 +27,17 @@ Phone: 743 and below
+    overflow: hidden;
   }
 
-  #photo-3 {
```
Giving a calls and id the same name might be confusing

Commit:
Fixed rendering bug for carousel
- Removed forcePhotoUpdate function. No longer needed
- Set up default image until real images are loaded

insivika reviewed (approved):
Looks good.
